quipucords/scanner/network/tests_network_inspect.py

```diff
@@ -284,83 +186,64 @@ def test_scan_inventory_grouping(self):
         scanner.connect_scan_task = self.connect_scan_task
-        with self.assertRaises(AnsibleError):
+        with self.assertRaises(AnsibleRunnerException):
             scanner._inspect_scan(Value('i', ScanJob.JOB_RUN), self.host_list)
             mock_run.assert_called()
 
-    @patch('scanner.network.inspect.InspectTaskRunner._inspect_scan',
-           side_effect=mock_scan_error)
+    @patch('scanner.network.inspect.InspectTaskRunner._inspect_scan')
     def test_inspect_scan_error(self, mock_scan):
         """Test scan flow with mocked manager and failure."""
-        scanner = InspectTaskRunner(
-            self.scan_job, self.scan_task)
+        mock_scan.side_effect = AnsibleRunnerException()
+        scanner = InspectTaskRunner(self.scan_job, self.scan_task)
         scan_task_status = scanner.run(Value('i', ScanJob.JOB_RUN))
         mock_scan.assert_called_with(ANY, self.host_list)
         self.assertEqual(scan_task_status[1], ScanTask.FAILED)
 
-    @patch('scanner.network.utils.TaskQueueManager.run',
-           side_effect=mock_run_success)
+    @patch('ansible_runner.run')
     def test_inspect_scan_fail_no_facts(self, mock_run):
         """Test running a inspect scan with mocked connection."""
-        expected = ([('1.2.3.4', {'name': 'cred1'})], [])
-        mock_run.return_value = expected
+        mock_run.return_value.status = 'successful'
         with requests_mock.Mocker() as mocker:
             mocker.post(self.fact_endpoint, status_code=201, json={'id': 1})
-            scanner = InspectTaskRunner(
-                self.scan_job, self.scan_task)
+            scanner = InspectTaskRunner(self.scan_job, self.scan_task)
+            scanner.connect_scan_task = self.connect_scan_task
             scan_task_status = scanner.run(Value('i', ScanJob.JOB_RUN))
-            mock_run.assert_called_with(ANY)
+            mock_run.assert_called()
             self.assertEqual(scan_task_status[1], ScanTask.FAILED)
 
-    def test_populate_callback(self):
-        """Test the population of the callback object for inspect scan."""
-        callback = InspectResultCallback(
-            scan_task=self.scan_task)
-        host = Mock()
-        host.name = '1.2.3.4'
-        task = Mock(args={'_raw_params': 'command line'})
-        result = Mock(_host=host, _result={'rc': 3}, _task=task)
-
-        callback.v2_runner_on_unreachable(result)
-
     def test_ssh_crash(self):
         """Simulate an ssh crash."""
         scanner = InspectTaskRunner(
             self.scan_job, self.scan_task)
         path = os.path.abspath(
             os.path.join(os.path.dirname(__file__),
-                         '../../../test_util/crash.py'))
+                         'test_util/crash.py'))
         _, result = scanner._inspect_scan(
             Value('i', ScanJob.JOB_RUN),
             self.host_list,
             base_ssh_executable=path)
-        self.assertEqual(result, ScanTask.COMPLETED)
```

Comment:
I think this test used to be fake. If we encountered a crash why would we want to have the scanTask report as Complete?